File: C#.net/MTS.ICLViewer/frmOpt.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;

namespace MTS.ICLViewer
{
	public partial class FrmOpt
	{

		internal FrmOpt()
		{
			InitializeComponent();
		}
	  private void OK_Button_Click(object sender, System.EventArgs e)
	  {
		// save options in settings
		Properties.Settings.Default.color01 = btnColor01.BackColor;
		Properties.Settings.Default.color10 = btnColor10.BackColor;
		Properties.Settings.Default.color20 = btnColor20.BackColor;
		Properties.Settings.Default.color25 = btnColor25.BackColor;
		Properties.Settings.Default.checkImage = cbCheckImage.Checked;
		Properties.Settings.Default.frontAndBack = cbFrontBack.Checked;
		Properties.Settings.Default.showItemNumber = cbItemNumber.Checked;
		Properties.Settings.Default.ClosePrompt = cbClosePrompt.Checked;
		this.DialogResult = System.Windows.Forms.DialogResult.OK;
		MessageBox.Show("Settings will be effective next time you open a file", "Options save", MessageBoxButtons.OK, MessageBoxIcon.Information);
        Properties.Settings.Default.Save();
		this.Close();
	  }

	  private void Cancel_Button_Click(object sender, System.EventArgs e)
	  {
		this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
		this.Close();
	  }

	  private void frmOpt_Load(object sender, System.EventArgs e)
	  {
		LoadSettings();
	  }

	  private void btnColor01_Click(object sender, System.EventArgs e)
	  {
		if (colorRec.ShowDialog() == System.Windows.Forms.DialogResult.OK)
		{
		  btnColor01.BackColor = colorRec.Color;
		}
	  }

	  private void btnColor10_Click(object sender, System.EventArgs e)
	  {
		if (colorRec.ShowDialog() == System.Windows.Forms.DialogResult.OK)
		{
		  btnColor10.BackColor = colorRec.Color;
		}
	  }

	  private void btnColor20_Click(object sender, System.EventArgs e)
	  {
		if (colorRec.ShowDialog() == System.Windows.Forms.DialogResult.OK)
		{
		  btnColor20.BackColor = colorRec.Color;
		}
	  }

	  private void btnColor25_Click(object sender, System.EventArgs e)
	  {
		if (colorRec.ShowDialog() == System.Windows.Forms.DialogResult.OK)
		{
		  btnColor25.BackColor = colorRec.Color;
		}
	  }

	  private void btnReset_Click(object sender, System.EventArgs e)
	  {
		Properties.Settings.Default.Reset();
		LoadSettings();
	  }

	  private void LoadSettings()
	  {
		// fill color options
		btnColor01.BackColor = Properties.Settings.Default.color01;
		btnColor10.BackColor = Properties.Settings.Default.color10;
		btnColor20.BackColor = Properties.Settings.Default.color20;
		btnColor25.BackColor = Properties.Settings.Default.color25;

		// fill display options
		cbCheckImage.Checked = Properties.Settings.Default.checkImage;
		cbFrontBack.Checked = Properties.Settings.Default.frontAndBack;
		cbItemNumber.Checked = Properties.Settings.Default.showItemNumber;
		cbClosePrompt.Checked = Properties.Settings.Default.ClosePrompt;
	  }

      private void tpColor_Click(object sender, EventArgs e)
      {

      }
	}

} //end of root namespace

File: VB.net/frmOpt.vb
Imports System.Windows.Forms

Public Class frmOpt

  Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click
    ' save options in settings
    My.Settings.color01 = btnColor01.BackColor
    My.Settings.color10 = btnColor10.BackColor
    My.Settings.color20 = btnColor20.BackColor
    My.Settings.color25 = btnColor25.BackColor
    My.Settings.checkImage = cbCheckImage.Checked
    My.Settings.frontAndBack = cbCheckImage.Checked
    My.Settings.showItemNumber = cbItemNumber.Checked
    My.Settings.ClosePrompt = cbClosePrompt.Checked
    Me.DialogResult = System.Windows.Forms.DialogResult.OK
    MessageBox.Show("Settings will be effective next time you open a file", "Options save", MessageBoxButtons.OK, MessageBoxIcon.Information)
    Me.Close()
  End Sub

  Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click
    Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
    Me.Close()
  End Sub

  Private Sub frmOpt_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
    LoadSettings()
  End Sub

  Private Sub btnColor01_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnColor01.Click
    If colorRec.ShowDialog() = Windows.Forms.DialogResult.OK Then
      btnColor01.BackColor = colorRec.Color
    End If
  End Sub

  Private Sub btnColor10_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnColor10.Click
    If colorRec.ShowDialog() = Windows.Forms.DialogResult.OK Then
      btnColor10.BackColor = colorRec.Color
    End If
  End Sub

  Private Sub btnColor20_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnColor20.Click
    If colorRec.ShowDialog() = Windows.Forms.DialogResult.OK Then
      btnColor20.BackColor = colorRec.Color
    End If
  End Sub

  Private Sub btnColor25_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnColor25.Click
    If colorRec.ShowDialog() = Windows.Forms.DialogResult.OK Then
      btnColor25.BackColor = colorRec.Color
    End If
  End Sub

  Private Sub btnReset_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnReset.Click
    My.Settings.Reset()
    LoadSettings()
  End Sub
  Private Sub LoadSettings()
    ' fill color options
    btnColor01.BackColor = My.Settings.color01
    btnColor10.BackColor = My.Settings.color10
    btnColor20.BackColor = My.Settings.color20
    btnColor25.BackColor = My.Settings.color25

    ' fill display options
    cbCheckImage.Checked = My.Settings.checkImage
    cbFrontBack.Checked = My.Settings.frontAndBack
    cbItemNumber.Checked = My.Settings.showItemNumber
    cbClosePrompt.Checked = My.Settings.ClosePrompt
  End Sub
End Class
